app/tests/ble/central/src/main.c

```diff
@@ -59,6 +63,26 @@ static void ble_central_native_posix_options(void) {
          .type = 'b',
          .dest = (void *)&clear_bond_on_disconnect,
          .descript = "Clear bonds on disconnect and reconnect"},
+        {.is_switch = true,
+         .option = "skip_set_security_on_connect",
+         .type = 'b',
+         .dest = (void *)&skip_set_security_on_connect,
+         .descript = "Skip set security level after connecting"},
+        {.is_switch = true,
+         .option = "read_hid_report_on_connect",
+         .type = 'b',
+         .dest = (void *)&read_hid_report_on_connect,
+         .descript = "Read the peripheral HID report after connecting"},
+        {.is_switch = true,
+         .option = "skip_discovery_on_connect",
+         .type = 'b',
+         .dest = (void *)&skip_discovery_on_connect,
+         .descript = "Skip GATT characteristic discovery after connecting"},
+        {.is_switch = true,
+         .option = "read_directly_on_discovery",
+         .type = 'b',
+         .dest = (void *)&read_directly_on_discovery,
+         .descript = "Read HIDS report after GATT characteristic discovery"},
```

Comment:
Add a test for peripheral initiated encryption setup when a bonded central does not encrypt after connection; i.e. scenarios where notifications were enabled in the GATT server in a previous connection with the central.

Reply:
So... After reviewing the code in Zephyr's `bt_gatt_connected`, that happens *only* if we have AUTO_SEC_REQ set. It would seem that Apple guidelines state that you should *not* do this automatically, seemingly even for existing bonded peers, so perhaps we should *only* do this *on attempting to notify* and we see that we still haven't had our connection encrypted yet. That *should* cover our use cases and still have better macOS compat?

Reply:
> > we should only do this on attempting to notify and we see that we still haven't had our connection encrypted yet. That should cover our use cases and still have better macOS compat?
> 
> Yes, only send SMP Security Req PDU when already bonded to the peer and not already encrypted in the connection to the required security level. Also, not to accept any new pairing but restrict to only initiating encryption using the existing keys. If the encryption fails, it should be regarded as a spoof device (that may have used the public Bluetooth address of the genuine central or somehow generated a private address of that genuine central to establish the connection).

Yup, this effectively does this since we only try to upgrade that connection if it's one from a host address stored in the active/selected peripheral slot, and the call to set_security under the hood sends the SMP security request PDU for us.